### ISSUE TITLE
Why is account name starting with a digit invalid?

### ISSUE THREAD

**Issue by blais** [beancount/beancount#245]
**[Original report](https://bitbucket.org/blais/beancount/issue/245) by Martin Michlmayr (Bitbucket: [tbm13](https://bitbucket.org/tbm13), GitHub: [tbm](https://github.com/tbm)).**

----------------------------------------

With:

```
Assets:Vouchers:99Ranch
```

I get:

```
syntax error, unexpected COLON, expecting EOL or COMMENT
```

I think this is a big unclear.  It should complain about the digits, not the colon.

**Comment by blais** [comment]
**Original comment by Metin Akat (Bitbucket: [m_akat](https://bitbucket.org/m_akat), GitHub: [loxs](https://github.com/loxs)).**

----------------------------------------

What's the problem with this name being legal?

**Comment by blais** [comment]
**Original comment by Martin Michlmayr (Bitbucket: [tbm13](https://bitbucket.org/tbm13), GitHub: [tbm](https://github.com/tbm)).**

----------------------------------------

Good point. I was wondering about the same.

Should this be allowed, Martin?

**Comment by blais** [comment]
**Original comment by Martin Michlmayr (Bitbucket: [tbm13](https://bitbucket.org/tbm13), GitHub: [tbm](https://github.com/tbm)).**

----------------------------------------

Sorry, with current beancount I get the right error:

```
ValueError: Invalid account name: Assets:Vouchers:99Ranch
```

But I think Metin Akat's question stands.  Why is this name not legal?

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Thanks for filing this bug. It highlights a latent issue, whereby this is supposed to be accepted and the lexer does output an ACCOUNT token, but the further verification of the five account prefixes (done late, in Python, because it requires to read the parsed options) wasn't in sync with that change.
See beancount.core.account and beancount.parser.valid_account_regexp.
I'll fix this right awy.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Fixed #245: Synchronized the definition of a valid account regular expression between the lexer and the later enhanced version (in Python) which checks the configured prefixes.
